modules/regional-go-service/network-iam.tf

```diff
@@ -0,0 +1,37 @@
+// This file gramnts permissions for Cloud Run to access VPCs.
+// In particular, this is configured to support connecting to a shared VPC in another project,
+// but should be safe to apply in the same project.
+// See https://cloud.google.com/run/docs/configuring/shared-vpc-direct-vpc#set_up_iam_permissions for more details.
+
+data "google_project" "project" {
+  project_id = var.project_id
+}
+
+// Grant Cloud Run the service access to the network.
+resource "google_project_iam_member" "cloudrun_service_network_user" {
+  project = var.network_project
+  member  = "serviceAccount:service-${data.google_project.project.number}@serverless-robot-prod.iam.gserviceaccount.com"
+  role    = "roles/compute.networkUser"
+}
```

Comment:
There are no values of this specific to this invocation of the module, which tells me that this is likely the wrong place for it.

I don't even think `networking` is the right place for it, I suspect that setting this up in `chainguard-dev/infra` to enable Cloud Run to utilize cross-project networking is probably what we want.